File: MyMausam-2.0/backend/python/app/schemas/chatbot.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel

class ChatbotQueryRequest(BaseModel):
    query: str
    location: Optional[str] = "Ghaziabad"
    language: Optional[str] = "English"
    persona: Optional[str] = "health"

class ChatbotQueryResponse(BaseModel):
    answer: str
    voice_summary: str
    action_links: Optional[List[Dict[str, str]]] = None

File: MyMausam-2.0/packages/types/chatbot.ts
export interface ChatMessage {
  id: string;
  sender: "user" | "bot" | "system";
  text: string;
  timestamp: string;
  language?: string;
  audio_url?: string;
  suggestions?: string[];
}

export interface ChatQueryRequest {
  query: string;
  location?: string;
  language?: string;
  persona?: string;
}

export interface ChatQueryResponse {
  answer: string;
  voice_summary: string;
  action_links?: { title: string; route: string }[];
  context_used?: {
    location: string;
    temperature: number;
    condition: string;
  };
}
